internal/datadog/metrics_client.go

```diff
@@ -55,13 +62,17 @@ func (m *metricsClient) Count(name string, value int64, tags []string, _ float64
 	if err != nil {
 		return err
 	}
-	attr := attributeFromTags(tags)
+	attr := m.attributeFromTags(tags)
 	counter.Add(context.Background(), value, metric.WithAttributeSet(attr))
 	return nil
 }
 
-func attributeFromTags(tags []string) attribute.Set {
+func (m *metricsClient) attributeFromTags(tags []string) attribute.Set {
 	attr := make([]attribute.KeyValue, 0, len(tags))
```

Comment:
We can add one more since the total length is one more
```suggestion
	attr := make([]attribute.KeyValue, 0, len(tags) + 1)
```